src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java

```diff
@@ -0,0 +1,148 @@
+     */
+    private static final Supplier<Double> LEFT_CORRECTIONS =
+            corrections.addConstantDouble("left correction", 1);
+    private static final Supplier<Double> RIGHT_CORRECTIONS =
+            corrections.addConstantDouble("right correction", 1);
+
+    private final Encoder leftEncoder, rightEncoder;
+    private final ADXRS450_Gyro gyro;
+
+    /**
+     * Constants for a PID controller.
+     * Since they were made using {@code addConstantDouble}, they are constants relative to the code itself, but are
+     * still able to be changed via the shuffleboard.
+     */
+    private final Supplier<Double> kP = rootNamespace.addConstantDouble("kP", 0);
+    private final Supplier<Double> kI = rootNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kD = rootNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> TOLERANCE = rootNamespace.addConstantDouble("TOLERANCE", 0);
+    private final Supplier<Double> WAIT_TIME = rootNamespace.addConstantDouble("WAIT_TIME", 1);
+
+    private final PIDSettings pidSettings = new PIDSettings(kP, kI, kD, TOLERANCE, WAIT_TIME);
+
+    /**
+     * Constants for a feed forward controller More info in {@link FeedForwardController}.
+     * Since they were made using {@code addConstantDouble}, they are constants relative to the code itself, but are
+     * still able to be changed via the shuffleboard.
+     */
+    private final Supplier<Double> kS = rootNamespace.addConstantDouble("kS", 0);
+    private final Supplier<Double> kV = rootNamespace.addConstantDouble("kV", 0);
+    private final Supplier<Double> kA = rootNamespace.addConstantDouble("kA", 0);
+    private final Supplier<Double> kG = rootNamespace.addConstantDouble("kG", 0);
+
+    private final FeedForwardSettings ffSettings = new FeedForwardSettings(kS, kV, kA, kG);
+
+    private static Drivetrain instance;
+
+    public static Drivetrain getInstance() {
+        if (instance == null) {
+            instance = new Drivetrain();
+            instance.configureDashboard();
+        }
+        return instance;
+    }
+
+    private Drivetrain() {
+        super(
+                "drivetrain",
+                new BustedMotorControllerGroup(
+                        LEFT_CORRECTIONS,
+                        new WPI_TalonSRX(RobotMap.CAN.LEFT_TALON_1),
+                        new WPI_TalonSRX(RobotMap.CAN.LEFT_TALON_2)
+                ),
+                new BustedMotorControllerGroup(
+                        RIGHT_CORRECTIONS,
+                        new WPI_TalonSRX(RobotMap.CAN.RIGHT_TALON_1),
+                        new WPI_TalonSRX(RobotMap.CAN.RIGHT_TALON_2)
+                )
+        );
+        this.leftEncoder = new Encoder(RobotMap.DIO.LEFT_ENCODER_A, RobotMap.DIO.LEFT_ENCODER_B);
+        this.rightEncoder = new Encoder(RobotMap.DIO.RIGHT_ENCODER_A, RobotMap.DIO.RIGHT_ENCODER_B);
+        this.gyro = new ADXRS450_Gyro();
+    }
+
+    public double getLeftDistance() {
```

Comment:
have you configured distance per pulse?
this will return the pulse, because no distancePerPulse was given